minifi/minifi-commons/minifi-utils/src/main/java/org/apache/nifi/minifi/commons/status/util/PropertyUtil.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.minifi.commons.status.util;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+public class PropertyUtil {
+    private static final String DOT = ".";
+    private static final String UNDERSCORE = "_";
+    private static final String HYPHEN = "-";
+
+    private PropertyUtil() {
+
+    }
+
+    public static String resolvePropertyValue(String name, Map<?, ?> properties) {
+        return Optional.of(name)
+            .map(n -> getPropertyValue(n, properties))
+            .or(() -> Optional.of(name.toUpperCase())
+                .filter(uppercasedName -> !name.equals(uppercasedName))
+                .map(n -> getPropertyValue(n, properties)))
+            .orElse(null);
+    }
+
+    private static String getPropertyValue(String name, Map<?, ?> properties) {
+        return Stream.concat(
+                Stream.of(name).filter(properties::containsKey),
+                Stream.of(name.replace(DOT, UNDERSCORE), name.replace(HYPHEN, UNDERSCORE), name.replace(DOT, UNDERSCORE).replace(HYPHEN, UNDERSCORE))
+                    .distinct()
+                    .filter(candidate -> !name.equals(candidate) && properties.containsKey(candidate)))
+            .findFirst()
+            .map(properties::get)
+            .map(String::valueOf)
+            .orElse(null);
```

Comment:
You could keep it as optional here as well and flatmap on the caller side as needed.

Reply:
done